shared.c

```diff
@@ -274,6 +278,69 @@ static gchar *get_file_extension(const gchar *filename)
     return extension;
 }
 
+void squash_extract_inode_to_file(sqfs *fs, sqfs_inode inode, const gchar *dest)
+{
+    off_t bytes_already_read = 0;
+    sqfs_off_t bytes_at_a_time = 64*1024;
+    FILE * f;
+    f = fopen (dest, "w+");
+    if (f == NULL){
+        fprintf(stderr, "fopen error\n");
+        return;
+    }
+    while (bytes_already_read < inode.xtra.reg.file_size)
+    {
+        char buf[bytes_at_a_time];
+        if (sqfs_read_range(fs, &inode, (sqfs_off_t) bytes_already_read, &bytes_at_a_time, buf))
+            fprintf(stderr, "sqfs_read_range error\n");
+        fwrite(buf, 1, bytes_at_a_time, f);
+        bytes_already_read = bytes_already_read + bytes_at_a_time;
+    }
+    fclose(f);
+}
+
+gchar * gess_install_destination(const gchar *desktop_icon_value_original, const char *path, const char *md5)
```

Comment:
Should it mean "guess"?

Reply:
yes, sorry for the typos I should get a spell checker in my IDE

Reply:
I'm sending a new pr right away

Reply:
np, thanks